piet/src/gradient.rs

```diff
@@ -262,26 +288,52 @@ impl LinearGradient {
             stops: stops.to_vec(),
         }
     }
+
+    // maybe these should be public API? that was my original intention but I'm not
+    // sure there's a clear use, so keeping them private for now.
+    /// Generate a [`FixedLinearGradient`] by mapping points in the unit square
+    /// onto points in `rect`.
+    ///
+    /// [`FixedLinearGradient`]: struct.FixedLinearGradient.html
+    fn resolve(&self, rect: Rect) -> FixedLinearGradient {
+        FixedLinearGradient {
+            start: self.start.resolve(rect),
+            end: self.end.resolve(rect),
+            stops: self.stops.clone(),
+        }
+    }
 }
 
 impl RadialGradient {
-    /// Creates a simple `RadialGradient`. This gradient has the same `origin`
-    /// and `center`, and uses the `Fill` [`ScaleMode`]. These attributes can be
-    /// modified with the [`with_origin`] and [`with_scale_mode`] builder methods.
+    /// Creates a simple `RadialGradient`. This gradient has `origin` and `center`
+    /// set to `(0.5, 0.5)`, and uses the `Fill` [`ScaleMode`]. These attributes can be
+    /// modified with the [`with_center`], [`with_origin`],
+    /// and [`with_scale_mode`] builder methods.
     ///
     /// [`ScaleMode`]: enum.ScaleMode.html
+    /// [`with_center`]: struct.RadialGradient.html#method.with_center
     /// [`with_origin`]: struct.RadialGradient.html#method.with_origin
     /// [`with_scale_mode`]: struct.RadialGradient.html#method.with_scale_mode
-    pub fn new(center: UnitPoint, radius: f64, stops: impl GradientStops) -> Self {
+    pub fn new(radius: f64, stops: impl GradientStops) -> Self {
         RadialGradient {
-            center,
-            origin: center,
+            center: UnitPoint::CENTER,
+            origin: UnitPoint::CENTER,
             radius,
             stops: stops.to_vec(),
             scale_mode: ScaleMode::Fill,
         }
     }
 
+    /// A builder-style method for changing the center of the gradient.
+    ///
+    /// See the main [`RadialGradient`] docs for an explanation of center vs. origin.
```

Comment:
Might be worth mentioning that it leaves the origin unchanged. (In the common case of no offset, now we'll have to call both methods)